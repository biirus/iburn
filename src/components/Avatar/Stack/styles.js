export default ({ spacing, colors, }) => {
  return {
    root: {
      display: "flex",
      lineHeight: 1,
    },

    item: {
      position: "relative",
      marginRight: -spacing.unit,
    },

    inner: {
      border: `2px solid ${colors.n0}`,
    },
  };
};
