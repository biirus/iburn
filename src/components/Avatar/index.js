import React from "react";
import cn from "classnames";
import PropTypes from "prop-types";

// enhancers
import { enhance } from "./enhnace";

/**
 * The basic Avatar component that renders it's content or image
 * which URL passed by `src` prop
 */
class Avatar extends React.Component {
  static defaultProps = {
    src: null,
  };

  static propTypes = {
    /** Image source URL */
    src: PropTypes.string,
  };

  clearProps = () => {
    const {
      classes,
      className,
      children,
      src,
      sheet,
      theme,
      ...rest
    } = this.props;

    return rest;
  };

  render () {
    const { classes, className, children, src, } = this.props;
    const classNames = cn(classes.root, className);

    return (
      <div {...this.clearProps()} className={classNames}>
        {src ? (
          <div
            className={classes.img}
            style={{ backgroundImage: `url(${src})`, }}
          />
        ) : (
          children
        )}
      </div>
    );
  }
}

export default enhance(Avatar);
