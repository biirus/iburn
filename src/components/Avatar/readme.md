### String children

```js
<Avatar>IB</Avatar>
<Avatar>AB</Avatar>
```

### Icon children

```js
<Avatar><Icon label="atlassian" /></Avatar>
<Avatar><Icon label="user-avatar-circle" /></Avatar>
```
