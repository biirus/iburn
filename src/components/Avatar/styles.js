export default ({ colors, typography, spacing }) => {
  const sizes = {
    xSmall: spacing.unit * 2,
    small: spacing.unit * 3,
    medium: spacing.unit * 4,
    large: spacing.unit * 5,
    xLarge: spacing.unit * 12,
    xxLarge: spacing.unit * 16,
  };

  const fontSizes = {
    xSmall: 6,
    small: 10,
    medium: 14,
    large: 18,
    xLarge: 40,
    xxLarge: 56,
  };

  return {
    /* Styles applied to the root element. */
    root: {
      color: colors.w700a,

      display: "inline-flex",
      alignItems: "center",
      justifyContent: "center",
      overflow: "hidden",
      width: ({ size }) => sizes[size],
      height: ({ size }) => sizes[size],

      fontFamily: typography.body.fontFamily,
      lineHeight: 1,
      fontSize: ({ size }) => fontSizes[size],

      backgroundColor: ({ color }) => color || colors.n100,
      borderRadius: "50%",
    },

    /* Styles applied to the img element if either `src` or `srcSet` is defined. */
    img: {
      width: "100%",
      height: "100%",
      objectFit: "cover",
    },
  };
};
