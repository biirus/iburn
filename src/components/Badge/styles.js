export default ({ spacing, palette, colors, borderRadius }) => {
  return {
    /* Styles applied to the root element */
    root: {
      display: "inline-flex",
      padding: [[spacing.small / 2, spacing.small]],
      lineHeight: 1,
      borderRadius,
    },

    /* Styles applied to the root element if `color="default"` */
    default: {
      backgroundColor: colors.n20a,
      color: colors.n400,
    },

    /* Styles applied to the root element if `color="primary"` */
    primary: {
      backgroundColor: palette.primary,
      color: palette.text.invert,
    },

    /* Styles applied to the root element if `color="error"` */
    error: {
      backgroundColor: palette.error,
      color: palette.text.invert,
    },

    /* Styles applied to the root element if `color="success"` */
    success: {
      backgroundColor: palette.success,
      color: palette.text.invert,
    },

    /* Styles applied to the root element if `color="warning"` */
    warning: {
      backgroundColor: palette.warning,
      color: palette.text.invert,
    },

    /* Styles applied to the root element if `color="info"` */
    info: {
      backgroundColor: palette.info,
      color: palette.text.invert,
    },
  };
};
