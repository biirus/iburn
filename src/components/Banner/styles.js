export default ({ spacing, palette, typography, zIndex, transitions }) => {
  return {
    root: {
      ...typography.h100,

      position: "fixed",
      top: 0,
      left: 0,
      width: "100%",
      zIndex: zIndex.flag,

      display: "flex",
      alignItems: "center",
      justifyContent: "center",
      padding: spacing.medium,

      opacity: 0,
      transform: `translateY(-100%)`,

      transition: transitions.create(["opacity", "transform"], "leavingScreen")
    },

    open: {
      opacity: 1,
      transform: `translateY(0)`,

      transition: transitions.create(["opacity", "transform"], "enteringScreen")
    },

    warning: {
      backgroundColor: palette.messages.warning,
      color: palette.text.invert,

      "& $icon": {
        fill: palette.messages.warning
      }
    },

    error: {
      backgroundColor: palette.messages.error,
      color: palette.text.invert,

      "& $icon": {
        fill: palette.messages.error
      }
    },

    info: {
      backgroundColor: palette.messages.info,
      color: palette.text.invert,

      "& $icon": {
        fill: palette.messages.info
      }
    },

    success: {
      backgroundColor: palette.messages.success,
      color: palette.text.invert,

      "& $icon": {
        fill: palette.messages.success
      }
    },

    icon: {
      flex: `0 0 ${spacing.unit * 3}px`
    },

    label: {
      flex: `0 1 auto`,
      marginLeft: spacing.small / 2,
      overflow: "hidden",
      textOverflow: "ellipsis",
      whiteSpace: "nowrap"
    }
  };
};
