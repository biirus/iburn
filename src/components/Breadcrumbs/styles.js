export default theme => {
  return {
    root: {
      overflow: "hidden",
      whiteSpace: "nowrap",
      textOverflow: "ellipsis",
      color: theme.palette.text.secondary,
    },

    link: {
      color: theme.palette.text.secondary,
      textDecoration: "none",

      "&:hover": {
        textDecoration: "underline",
      },
    },

    item: {
      "&+$item::before": {
        content: `"/"`,
        display: "inline-block",
        width: theme.spacing.medium,
        textAlign: "center",
      },
    },
  };
};
