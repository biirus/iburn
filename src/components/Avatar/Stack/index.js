import React from "react";
import cn from "classnames";
import PropTypes from "prop-types";

// enhancers
import withStyles from "react-jss";

// styles
import styles from "./styles";

class AvatarsStack extends React.Component {
  static propTypes = {
    children: PropTypes.arrayOf(PropTypes.element),
  };

  render () {
    const { className, classes, children, sheet, theme, ...rest } = this.props;
    const classNames = cn(classes.root, className);

    return (
      <div {...rest} className={classNames}>
        {React.Children.map(children, (child, index) => {
          return (
            <div
              key={index}
              className={classes.item}
              style={{ zIndex: children.length - index, }}
            >
              {React.cloneElement(child, { className: classes.inner, })}
            </div>
          );
        })}
      </div>
    );
  }
}

const enhance = withStyles(styles);

export default enhance(AvatarsStack);
