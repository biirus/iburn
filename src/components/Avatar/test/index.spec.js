import React from "react";

import Avatar from "../index";
import styles from "../styles";

const componentClasses = classes(styles);
const Component = Avatar.InnerComponent;

describe("<Avatar />", () => {
  it("should render Avatar", () => {
    const wrapper = shallow(
      <Component classes={componentClasses}>IB</Component>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("should compose passed class names", () => {
    const actual = shallow(
      <Component classes={componentClasses} className="customClass">
        IB
      </Component>
    );

    expect(actual.prop("className")).toBe("root customClass");
  });

  it("should render image instead of children if `src` prop is passed", () => {
    const wrapper = shallow(
      <Component classes={componentClasses} src="https://google.com">
        IB
      </Component>
    );

    expect(wrapper.find(".img").exists()).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });

  it("should render image instead children if `src` prop is not passed", () => {
    const wrapper = shallow(
      <Component classes={componentClasses}>IB</Component>
    );

    expect(wrapper.find(".img").exists()).toBeFalsy();
    expect(wrapper).toMatchSnapshot();
  });
});
