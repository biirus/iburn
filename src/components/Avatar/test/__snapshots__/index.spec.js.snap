// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> should render Avatar 1`] = `
<div
  className="root"
>
  <div
    className="wrapper"
  >
    IB
  </div>
</div>
`;

exports[`<Avatar /> should render image instead children if \`src\` prop is not passed 1`] = `
<div
  className="root"
>
  <div
    className="wrapper"
  >
    IB
  </div>
</div>
`;

exports[`<Avatar /> should render image instead of children if \`src\` prop is passed 1`] = `
<div
  className="root"
>
  <div
    className="wrapper"
  >
    <div
      className="img"
      style={
        Object {
          "backgroundImage": "url(https://google.com)",
        }
      }
    />
  </div>
</div>
`;
